knora/dsp_tools.py

```diff
@@ -233,11 +237,28 @@ def program(user_args: list[str]) -> None:
                   shortcode=args.shortcode,
                   default_ontology=args.default_ontology)
     elif args.action == 'start-api' and not sys.platform.startswith('win'):
-        output = subprocess.run(['/bin/bash', os.path.join(current_dir, 'dsplib/utils/start-api.sh')])
+        try:
+            response = requests.get("https://raw.githubusercontent.com/dasch-swiss/dsp-api/main/.github/actions/preparation/action.yml")
+            action = yaml.safe_load(response.content)
+            for step in action.get("runs", {}).get("steps", {}):
+                if re.search("(JDK)|(Java)", step.get("name", "")):
+                    distribution = step.get("with", {}).get("distribution", "").lower()
+                    java_version = step.get("with", {}).get("java-version", "").lower()
+        except requests.ConnectionError:
+            distribution = "temurin"
+            java_version = "17"
```

Comment:
for my taste this goes too far: I understand that not everyone is a developer. But keeping up with the right Java version is not too much to ask. And I suspect we change the names of our Github Actions more frequently than our Java versions, so this script would break less if you'd just hard-code temurin 17 here

Reply:
I made sure it will never break by making the `except` clause bare (not recommended, I know. But it will serve the purpose here)